File: moderate/m2last.clj
(require '[clojure.string :as str])
(doseq [item (for [st (str/split-lines (slurp (first *command-line-args*)))]
  (#(when true
      (def l (Integer/parseInt (last %)))
      (if (< l (count %))
        (nth % (- (count %) (inc l)))
        nil))
(str/split st #" ")))]
  (if (not (empty? item))
    (println item)))

File: moderate/m2last.lua
for line in io.lines(arg[1]) do
  local b = {}
  for i in string.gmatch(line, "%S+") do
    b[#b + 1] = i
  end
  local i = #b - b[#b]
  if i > 0 then
    print(b[i])
  end
end
